### ISSUE TITLE
Incorrect interface OktaVueOptions

### ISSUE THREAD

**Issue by garretmh** [okta/okta-vue#121]
### Describe the bug?

The `OktaVueOptions` interface is incorrect. The `onAuthRequired` and `onAuthResume` properties are both marked as required but should be optional.

```diff
export interface OktaVueOptions {
  oktaAuth: OktaAuth;
- onAuthRequired: OnAuthRequiredFunction;
- onAuthResume: OnAuthRequiredFunction;
+ onAuthRequired?: OnAuthRequiredFunction;
+ onAuthResume?: OnAuthRequiredFunction;
}
```

### What is expected to happen?

According to the docs, using the OktaVue plugin with just oktaAuth defined should work.

### What is the actual behavior?

In TypeScript projects, when using the OktaVue plugin both `onAuthRequired` and `onAuthResume` are required.

### Reproduction Steps?

1. Create a Vue project with TypeScript.
1. Follow the [configuration from the README](https://github.com/okta/okta-vue#configuration):

```ts
// main.ts

import { createApp } from 'vue'
import { OktaAuth } from '@okta/okta-auth-js'
import OktaVue from '@okta/okta-vue'

const oktaAuth = new OktaAuth({
  issuer: 'https://{yourOktaDomain}/oauth2/default',
  clientId: '{clientId}',
  redirectUri: window.location.origin + '/login/callback',
  scopes: ['openid', 'profile', 'email']
})

const app = createApp(App)
app.use(OktaVue, { oktaAuth })
app.mount('#app')
```

### SDK Versions

- `@okta/okta-auth-js` 6.9.0
- `@okta/okta-vue` 5.5.0
- `typescript` 4.9.0
- `vue` 3.2.47

### Execution Environment

- Node: 18

**Comment by denysoblohin-okta** [comment]
Thanks for submitting this issue.
Internal ref: OKTA-575263

**Comment by Ancient-Dragon** [comment]
Any updates to this? Currently facing this issue as well

**Comment by tenyo** [comment]
I ran into the same issue and was able to narrow it down to `vue` v3.2.46 (https://github.com/vuejs/core/releases/tag/v3.2.46) - hope that helps.

For now my workaround is:
```
"vue": "=3.2.45"
```

**Comment by Ancient-Dragon** [comment]
You can also just use:
```
app.use(OktaVue, { oktaAuth } as any)
```

That way you can continue to use the latest version of vue.

**Comment by jaredperreault-okta** [comment]
Btw, this was fixed in v `5.6.0`
